src/files-and-videos/videos/data/thunks.js

```diff
@@ -0,0 +1,352 @@
+        },
+      }));
+      dispatch(updateEditStatus({ editType: 'usageMetrics', status: RequestStatus.SUCCESSFUL }));
+    } catch (error) {
+      dispatch(updateErrors({ error: 'usageMetrics', message: `Failed to get usage metrics for ${video.displayName}.` }));
+      dispatch(updateEditStatus({ editType: 'usageMetrics', status: RequestStatus.FAILED }));
+    }
+  };
+}
+
+export function fetchVideoDownload({ selectedRows }) {
+  return async (dispatch) => {
+    dispatch(updateEditStatus({ editType: 'download', status: RequestStatus.IN_PROGRESS }));
+    const errors = await getDownload(selectedRows);
+    if (isEmpty(errors)) {
+      dispatch(updateEditStatus({ editType: 'download', status: RequestStatus.SUCCESSFUL }));
+    } else {
+      errors.forEach(error => {
+        dispatch(updateErrors({ error: 'download', message: error }));
+      });
+      dispatch(updateEditStatus({ editType: 'download', status: RequestStatus.FAILED }));
+    }
+  };
+}
+
+export function clearAutomatedTranscript({ courseId }) {
+  return async (dispatch) => {
+    dispatch(updateEditStatus({ editType: 'transcript', status: RequestStatus.IN_PROGRESS }));
+
+    try {
+      await deleteTranscriptPreferences(courseId);
+      dispatch(updateEditStatus({ editType: 'transcript', status: RequestStatus.SUCCESSFUL }));
+    } catch (error) {
+      dispatch(updateErrors({ error: 'transcript', message: 'Failed to update order transcripts settings.' }));
+      dispatch(updateEditStatus({ editType: 'transcript', status: RequestStatus.FAILED }));
+    }
+  };
+}
+
+export function updateTranscriptCredentials({ courseId, data }) {
+  return async (dispatch) => {
+    dispatch(updateEditStatus({ editType: 'transcript', status: RequestStatus.IN_PROGRESS }));
+
+    try {
+      await setTranscriptCredentials(courseId, data);
+      dispatch(updateTranscriptCredentialsSuccess({ provider: camelCase(data.provider) }));
+      dispatch(updateEditStatus({ editType: 'transcript', status: RequestStatus.SUCCESSFUL }));
+    } catch (error) {
+      dispatch(updateErrors({ error: 'transcript', message: `Failed to update ${data.provider} credentials.` }));
+      dispatch(updateEditStatus({ editType: 'transcript', status: RequestStatus.FAILED }));
+    }
+  };
+}
+
+export function updateTranscriptPreference({ courseId, data }) {
+  return async (dispatch) => {
+    dispatch(updateEditStatus({ editType: 'transcript', status: RequestStatus.IN_PROGRESS }));
+
+    try {
+      const preferences = await setTranscriptPreferences(courseId, data);
+      dispatch(updateTranscriptPreferenceSuccess(preferences));
+      dispatch(updateEditStatus({ editType: 'transcript', status: RequestStatus.SUCCESSFUL }));
+    } catch (error) {
+      dispatch(updateErrors({ error: 'transcript', message: `Failed to update ${data.provider} transcripts settings.` }));
+      dispatch(updateEditStatus({ editType: 'transcript', status: RequestStatus.FAILED }));
+    }
+  };
+}
```

Comment:
Probably there should be tests for everything here? I get that it's a lot. What's the alternative? We have a lot of complex code, so we want everything tested well, right?